### DISCUSSION TITLE
[Feature] Allow option to hide all photos w/ a given face

### DISCUSSION THREAD

**Discussion by joestump** [immich-app/immich#5480] [Feature Request]
### The feature

Currently, when I mark a face as hidden photos of that person will still show up in things like "Places", favorites, etc. I would like the option to hide all photos with a given face in it. 

Alternatively, the option to auto-archive all photos with a given face would be nice.

### Platform

- [ ] Server
- [X] Web
- [ ] Mobile

**Comment by varpness** [comment]
Not a fan of that person, eh?

**Reply by d0nk0tr0n** [reply]
Not a fan of constructive contribution, eh?

**Comment by Theprim0** [comment]
Would be a nice feature +1

**Comment by dob323** [comment]
I second this

**Comment by Sblop** [comment]
Perhaps an alternative approach could involve archiving all photos in which this individual appears? However, such a feature, if incorporated, should be in the settings for each person. Eg where set birthdays, block photos featuring the individual, and so forth.

I must admit, this at first seems like a rather niche scenario. However let's consider the most plausible situation: harboring animosity towards an ex-wife. By blocking all photos containing her, you inadvertently exclude a significant portion of images featuring your children as well. It raises the question whether this issue is best addressed through software complexity or through personal management of relationships. Just a thought. :o) I have a number of people i loathe in my gallery, however this is life. I much prefer the timeline, and I am able to see a picture of a person I do not like, without frowning. Comes with age. :o)

**Reply by Kernald** [reply]
It goes beyond personal feelings. Your current partner might not love seeing photos of you with your ex.

**Reply by LesKent** [reply]
I agree that adding a feature to archive all pictures that contain the specific person in the options for the person would be awesome. While I don't have a problem seeing old pictures of my toxic ex per se, I would love to share from my 25 year photo collection with my kids, family and friends without her in them so they don't have an re-live many unpleasant memories. 

I think the current "Hide Person" feature should stay the same, which in my view keeps the all "People View" from being cluttered with faces from strangers as it is often the case when taking multiple pictures at large crowd events, such as school and sports events or in public places.

**Reply by varpness** [reply]
I disagree. Maybe for you it will exclude a bunch of pictures of your children, but for others it won't. Every person has a different situation. Some people are camera shy and aren't in all the pictures with the kids. Applying your anecdotes to tell others how they should organize their own personal photo libraries doesn't work.

Also, suggesting your approach is better because you think you're older? C'mon now.

**Reply by LesKent** [reply]
I just shared by use case I believe others would also have. My perspective comes from someone with a large photo library of 120K pictures with a very long time frame that has unique challenges than is maybe typical for someone with less history. For example geotagging was rare until 2007, until cell phones started being used to take photos. Even today many SLR cameras still don't have geotagging capabilities. Over half my photos have no geolocation information. Current state of the art for facial recognition has challenges in identifying faces that evolve from babies to adults, and more so identifying differences between faces of very young siblings. Picking thresholds for minimum detection, minimum distance, and groupings leads to unwanted side effects such as having many strangers being identified in your people list. Mine has 1,385 yet I really only know about 300 of the people in my photos. The person I would most likely want to auto archive is identified on 2,096 assets over 15 years. Archiving that many manually with the current user interface would be very time consuming effort. Immich is amazing for what it does at this point and I hope to see it evolve and improve to make it easier to manage large libraries with scalable capabilities for large groupings.

**Reply by janexner** [reply]
Agree with @Kernald, and I think over time this is not a niche use case for anybody.

Maybe you need to hide people because your current surroundings don't appreciate, maybe you had a break-up or other traumatic experience, maybe people just drifted away and are no longer "memory-worthy".

For me, personally, I would love it right now because of a break-up and the associated trauma.

**Comment by woutervanelten** [comment]
Would like this option also.
Like mentioned before, under the option per person, where you also set the birthday, have a option to hide (or archive) all photo's containing this person and the person in "people".

**Comment by acknologia** [comment]
I came to request the same feature.

**Comment by magicreconpi** [comment]
I came here looking for this feature, but found a good enough work around. I selected all photo's with a given face and moved them to the locked folder. now they disappear in the normal list and you can only see them with a pin

**Comment by mynameismaurizio** [comment]
I'd even like to not see memories of images with the hidden faces get generated. Simple as that.

**Comment by lollo03** [comment]
I would like this feature

**Comment by giacorri** [comment]
+1 — and to help this move past "+1", here's a concrete, **opt-in and non-destructive** design that addresses the usual objections (why it's not "just archive", and the multi-person edge case).

**The idea:** reuse the existing `person.isHidden` flag and add one setting — *"Hide photos of hidden people from browse views"* — **off by default**. When on, assets containing a hidden person are filtered out of the **main browse surfaces** (Timeline, Places, Memories), while staying fully available in **People, search, albums, and the person's own page**.

**Why this isn't "just archive" (and why it dodges the #15339 problems):**
- It's a **view filter**, not a state change — it doesn't touch `AssetVisibility`, doesn't move files, and is instantly reversible by toggling the setting.
- The person stays in **People** and remains **searchable** — which is exactly what breaks today when you archive all of someone's photos (#15339). Nothing is lost.
- The "locked folder" workaround some mention here works, but it's manual, per-asset, and PIN-gated — not "set a person once, applies to all current + future photos".

**The one real design question — photos with multiple people** (a hidden person + someone you want to see):
- **A (any):** exclude if *any* face is a hidden person — matches the literal "hide all photos with this face".
- **B (all):** exclude only if *every* face is hidden.
- Suggested default **A**, because nothing is lost: the photo still lives in albums, search, and the visible person's page — it's just out of the daily timeline. Could be a 3-state setting (off / any / all) if preferred.

**Rough implementation (low-risk, additive):** the timeline/browse asset query gains an optional `NOT EXISTS` filter on `asset_faces → person.isHidden`, scoped to the owner and gated behind the new setting. No schema change, no migration, default behavior untouched. The timeline is a hot path, so it needs a supporting index and a benchmarked query.

I self-host and am happy to implement + benchmark this if a maintainer signals the opt-in, view-only approach is acceptable in principle. Posting the design here rather than opening yet another thread.

**Reply by Lauritz-Tieste** [reply]
Please correct me if I'm wrong, but isn't it possible to implement this feature using workflows? After all, if you set "On Upload" as the trigger, you can use a workflow to move the file to a locked folder. Just an idea.

**Reply by giacorri** [reply]
Good thought, but a few issues:

- "On Upload" fires before face recognition runs — the person is not identified yet at that point
- Existing photos (the main pain point for large libraries) would not be covered
- Moving to locked folder mutates visibility permanently; a toggle-based view filter is instantly reversible without touching any asset state

A person-aware workflow trigger could help for future uploads eventually, but the core ask is still a server-side filter on the browse query.
